packages/workflow/src/records/handler/reject.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors located at https://github.com/opencrvs/opencrvs-core/blob/master/AUTHORS.
+ */
+
+import * as z from 'zod'
+import { getLoggedInPractitionerResource } from '@workflow/features/user/utils'
+import { createRoute } from '@workflow/states'
+import { getToken } from '@workflow/utils/authUtils'
+import { validateRequest } from '@workflow/utils/index'
+import { toRejected } from '@workflow/records/state-transitions'
+import { sendBundleToHearth } from '@workflow/records/fhir'
+import { indexBundle } from '@workflow/records/search'
+import { auditEvent } from '@workflow/records/audit'
+
+const requestSchema = z.object({
+  comment: z.custom<string>(),
+  reason: z.custom<string>()
+})
+
+export const rejectRoute = createRoute({
+  method: 'POST',
+  path: '/records/{recordId}/reject',
+  allowedStartStates: ['READY_FOR_REVIEW', 'IN_PROGRESS', 'VALIDATED'],
+  action: 'REJECT',
+  handler: async (request, record) => {
+    const token = getToken(request)
+    const payload = validateRequest(requestSchema, request.payload)
+    const practitioner = await getLoggedInPractitionerResource(token)
+
+    const commentCode: fhir3.CodeableConcept = {
+      text: payload.comment
+    }
+
+    const { rejectedRecord, rejectedRecordWithTaskOnly } = await toRejected(
+      record,
+      practitioner,
+      commentCode,
+      payload.reason
+    )
+
+    await sendBundleToHearth(rejectedRecordWithTaskOnly)
+    await indexBundle(rejectedRecordWithTaskOnly, token)
```

Comment:
On second thought let's amend the tech debt that we have here, instead of just sending the task, let's forward the full record to search service and amend the logic there as necessary

Reply:
We can do the indexing of the bundle into the search service's work on a separate PR. It can be as is for now.